rfcs/0137-nix-language-version.md

```diff
@@ -0,0 +1,115 @@
+---
+feature: nix-language-version
+start-date: 2022-12-12
+author: @fricklerhandwerk
+co-authors: @thufschmitt @Ericson2314
+shepherd-team: 
+shepherd-leader: 
+related-issues: https://github.com/NixOS/nix/issues/7255
+---
+
+# Summary
+[summary]: #summary
+
+Introduce a convention to determine which version of the Nix language grammar to use for parsing a Nix file.
```

Comment:
A Nix file means that during the same evaluation different files can have different grammar versions, right?

Reply:
That is my interpretation. This would allow incremental evolution.

Reply:
IMO if it's declared per file then it's also meant per file. The alternative would be to have some special file in a project's root or something (like Rust and many other programming language projects do), but given Nix's rather YOLO attitude on imports I don't think that's feasible.

Whether projects want to enforce some version consistency across files however is a different question.